testing/test_text_layout.c

```diff
@@ -0,0 +1,116 @@
+#include "game/gui/text/text_layout.h"
+#include "resources/fonts.h"
+#include "utils/allocator.h"
+#include "utils/str.h"
+#include "video/surface.h"
+#include <CUnit/CUnit.h>
+
+void create_fake_font(font *font) {
+    font_create(font);
+    font->w = 8;
+    font->h = 8;
```

Comment:
Could you comment on what distinguishes the font width/height from the character surface width/height, or what they do?

Reply:
Old width and height are required only for the old renderer now, and are there for compatibility. I added comments to the font struct.